optimize/backend/src/test/java/io/camunda/optimize/service/util/BpmnModelUtilTest.java

```diff
@@ -35,72 +36,76 @@ public class BpmnModelUtilTest {
-    assertThat(serviceTaskflowNodeDataDto.getName()).isEqualTo(SERVICE_TASK);
-    assertThat(serviceTaskflowNodeDataDto.getType()).isEqualTo(SERVICE_TASK_TYPE);
-
-    FlowNodeDataDto startEventflowNodeDataDto = flowNodeData.get(2);
-    assertThat(startEventflowNodeDataDto.getId()).isNotNull();
-    assertThat(startEventflowNodeDataDto.getId()).isEqualTo(START_EVENT);
-    assertThat(startEventflowNodeDataDto.getName()).isEqualTo(START_EVENT);
-    assertThat(startEventflowNodeDataDto.getType()).isEqualTo(START_EVENT);
+
+    // then
+    assertThat(flowNodeData)
+        .isNotNull()
+        .isNotEmpty()
+        .extracting(FlowNodeDataDto::getId, FlowNodeDataDto::getType, FlowNodeDataDto::getName)
+        .containsExactlyInAnyOrder(
+            Tuple.tuple(START_EVENT, START_EVENT, START_EVENT),
+            Tuple.tuple(SERVICE_TASK, SERVICE_TASK_TYPE, SERVICE_TASK),
+            Tuple.tuple(END_EVENT, END_EVENT_TYPE, null));
   }
 
   @Test
   void shouldExtractUserTaskNames() {
+    // when
     final String bpmnModelInstance =
         Bpmn.convertToString(createSimpleUserTaskProcess(CUSTOMER_ONBOARDING));
     final Map<String, String> userTaskNames = BpmnModelUtil.extractUserTaskNames(bpmnModelInstance);
-    assertThat(userTaskNames).isNotNull();
-    assertThat(userTaskNames.isEmpty()).isFalse();
 
-    assertThat(userTaskNames.containsKey(USER_TASK)).isTrue();
+    // then
+    assertThat(userTaskNames).isNotNull().isNotEmpty().containsKey(USER_TASK);
   }
 
   @Test
   void shouldExtractProcessDefinitionName() {
+    // when
     final Optional<String> processName =
         BpmnModelUtil.extractProcessDefinitionName(
             CUSTOMER_ONBOARDING, SIMPLE_SERVICE_TASK_PROCESS);
-    assertThat(processName.isPresent()).isTrue();
+
+    // then
+    assertThat(processName).isPresent();
     assertThat(CUSTOMER_ONBOARDING).isEqualTo(processName.get());
   }
 
   @Test
   void shouldExtractFlowNodeNames() {
+    // when
     final List<FlowNodeDataDto> flowNodeData =
         BpmnModelUtil.extractFlowNodeData(SIMPLE_SERVICE_TASK_PROCESS);
     final Map<String, String> flowNodeNames = BpmnModelUtil.extractFlowNodeNames(flowNodeData);
-    assertThat(flowNodeNames).isNotNull();
-    assertThat(flowNodeNames.isEmpty()).isFalse();
 
-    assertThat(flowNodeNames.containsKey(START_EVENT)).isTrue();
-    assertThat(START_EVENT).isEqualTo(flowNodeNames.get(START_EVENT));
+    // then
+    assertThat(flowNodeNames)
+        .isNotNull()
+        .isNotEmpty()
+        .containsKey(START_EVENT)
+        .containsValue(START_EVENT);
   }
 
   @Test
   void shouldGetCollapsedSubprocessElementIds() {
+    // when
     final Set<String> collapsedSubprocessIds =
         BpmnModelUtil.getCollapsedSubprocessElementIds(SIMPLE_SERVICE_TASK_PROCESS);
-    assertThat(collapsedSubprocessIds).isNotNull();
-    assertThat(collapsedSubprocessIds.isEmpty()).isTrue();
+
+    // then
+    assertThat(collapsedSubprocessIds).isNotNull().isEmpty();
```

Comment:
I think the real value of this test would be extracting the actual events from a collapsed subprocess. There is a method in ZeebeModels called `createProcessWith83SignalEvents`, this should give you a process with a subprocess to work with

Reply:
fixed